### ISSUE TITLE
Can't find axisfifo under the /dev

### ISSUE THREAD

**Issue by markusxu** [jacobfeder/axisfifo#13]
I tried to use AXI_Stream_FIFO v4.2 IP core in Petalinux. I enabled the Xilinx AXI-Stream FIFO v4.1/v4.2 IP core driver directly in the petalinux kernel configuration, but after booting I did not find a device like /dev/axis_fifo__########. I'd like to ask what's going on, is there an operation I'm missing?

Board: ZCU104
Petalinux Version: 2020.2
Vivado Version: 2020.2

dmesg:
[dmesg.txt](https://github.com/jacobfeder/axisfifo/files/6397554/dmesg.txt)

pl.dtsi:
[pl.dtsi.txt](https://github.com/jacobfeder/axisfifo/files/6397583/pl.dtsi.txt)

vivado bd:
[top_design.pdf](https://github.com/jacobfeder/axisfifo/files/6397604/top_design.pdf)

**Comment by jacobfeder** [comment]
Usually this is a due to not connecting the IP block correctly in the block diagram. Make sure you have the interrupt lines connected.

**Comment by jacobfeder** [comment]
https://github.com/jacobfeder/axisfifo/issues/2

**Comment by markusxu** [comment]
I checked all my connections and it's now stuck at

[ 17.183168] axis-fifo driver loaded with parameters read_timeout = 1000, write_timeout = 1000

There is no output after this log

**Comment by jacobfeder** [comment]
Does the device show under /dev/axis_fifo_######## ? If so, try reading/writing it.

**Comment by markusxu** [comment]
No, in fact I can't even get into Linux.
By the way, I'm using version 2020.2, is there a version problem

**Comment by gutelfuldead** [comment]
> I checked all my connections and it's now stuck at
> 
> [ 17.183168] axis-fifo driver loaded with parameters read_timeout = 1000, write_timeout = 1000
> 
> There is no output after this log

If Linux freezes here it could because the AXI Clock isn't connected (or note enabled in the dts) and the module probe locks up when it tries to access the FPGA register via AXI4-Lite. I would make sure,

1. aclk and aresetn are connected correctly
2. fclk is enabled in device tree
3. you have the correct address for the logic in the dts entry

I've only tested on 2018.2 but I'd be surprised if this was caused by a version issue.

**Comment by markusxu** [comment]
I made a mistake and indeed there was no signal from the AXI clock.

**Comment by hemanthr28** [comment]
Hello, I am facing the same issue with my design. I have connected all the clocks and resets as required. When I load the module, the probe function is not called. I have tried adding printk's to verify this (none of the printk's in the probe function is executed). I only have the below output. 
I am using Petalinux 2020.2 and Vivado 2020.2.
dmesg output - 
[  104.140093] axis_fifo: loading out-of-tree module taints kernel.
[  104.147218] axis-fifo driver loaded with parameters read_timeout = 1000, write_timeout = 1000

I have attached the design for reference, any help would be appreciated. Thank you. 
[zynq_ps.pdf](https://github.com/jacobfeder/axisfifo/files/8219595/zynq_ps.pdf)
